File: src/decorator/pack.go
package decorator

type pack struct {
	desc string
	gift Goods
}

func NewPack(gift Goods, pack_desc string) pack {
	pack := pack{desc: pack_desc, gift: gift}
	return pack
}

func (pack pack) GetDesc() string {
	return pack.gift.GetDesc() + ", with " + pack.desc + " packed"
}

File: java/src/pattern/decorator/Pack.java
package pattern.decorator;

public class Pack extends Goods {
    private Goods goods;
    private String name;

    public Pack(String name) {
        this.name = name;
    }

    public void packet(Goods goods) {
        this.goods = goods;
    }

    @Override
    public String sold() {
        return this.name + "( " + this.goods.sold() + " )";
    }
}
